### PR TITLE
Adding max distance to plotting script

### PR DESCRIPTION
**Pull request by mariuswinkler**
Hi Hauke,

I redid what I mixed up in the other PR. In the end it's just a little function compution the distance travelled in the python plotting script.

One issue I couldn't resolve is that the function I created uses "from geopy.distance import geodesic" in the preamble for which one needs to have the package "geopy".
Where do I add "geopy" in the environement.yaml from which one installs pysonde at the very beginning when calling "conda install pysonde" ?

### CODE REVIEW

**Comment by mariuswinkler** [comment]
Sorry man. What I intended didn't work out. This PR contains all the other changes by Marius Rixen in respect to the .cor files as well and therefore overlaps with #27. I tried to only add the plotting stuff changes but since I did the PR from the forked repo it adds the .cor related changes by Marius Rixen as well. 

To seperate plotting from .cor related changes I would probably need to fork your orginal pysonde repo again right ?
I can also close this PR again, wait until you reviewed the .cor related changes and raise a clean PR afterwards.

What would be best ?

**Comment by observingClouds** [comment]
I would ask you to do a clean PR as the .cor PR might get changed a bit.

```
git checkout adding_max_distance_to_plotting_script
git reset --hard HEAD origin/master. # assuming origin is observingClouds as the master in your group RadioGouGou is modified
git cherry-pick 7609fc4
git push --force
```

are your friends again

File: scripts/plot_radiosonde_data.py [hunk 1]
```diff
@@ -112,6 +113,30 @@ def data4skewt(direction, data):
...
+    longitudes = ds['lon'].values
+    
+    # Ensure there are no NaN values by filtering them out
+    valid_indices = ~np.isnan(latitudes) & ~np.isnan(longitudes)
+    latitudes = latitudes[valid_indices]
+    longitudes = longitudes[valid_indices]
+    
+    # Skip soundings with insufficient valid data
+    if len(latitudes) < 2 or len(longitudes) < 2:
+        print(f"Not enough valid data for sounding {sounding}. Skipping.")
```
Comment (observingClouds) [inline scripts/plot_radiosonde_data.py:127]: `sounding` is not defined

File: scripts/plot_radiosonde_data.py [hunk 2]
```diff
@@ -112,6 +113,30 @@ def data4skewt(direction, data):
...
+        return None
+    
+    # Get the surface (initial) latitude and longitude
+    surface_lat = latitudes[0]
+    surface_lon = longitudes[0]
+    
+    # Calculate the distance from the surface point to each other point
+    distances = np.array([geodesic((surface_lat, surface_lon), (lat, lon)).kilometers 
+                          for lat, lon in zip(latitudes, longitudes)])
+
```
Comment (observingClouds) [inline scripts/plot_radiosonde_data.py:137]: Could you please run the pre-commit hooks. The formatting here does not comply with the set standard.

You can do:
```
pip install pre-commit
pre-commit install
pre-commit run --all
git add <the changed files>
git commit
```

**Review by observingClouds — COMMENTED** [review]
We are getting there, yes 🤣 Please have a look at my individual comments.

**Review by observingClouds — CHANGES_REQUESTED** [review]
I would also ask you to:
- add the geopy package to the `pyproject.toml` ([test currently fails)](https://github.com/observingClouds/pysonde/actions/runs/10884642525/job/30202955186?pr=33)
- describe your changes in the CHANGELOG.rst

**Review by observingClouds — APPROVED** [review]
Thanks for the addition and fixes! LGTM! Thanks for your first contribution! The tests are succeeding and the updated figure looks great.

track_2020-01-26T2244_ascent

**Pull request merged**